jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -741,7 +738,24 @@ ecma_string_to_utf8_string (const ecma_string_t *string_desc_p, /**< ecma-string
 
   JERRY_ASSERT (size <= buffer_size);
   return size;
-} /* ecma_string_to_utf8_string */
+} /* ecma_string_copy_to_utf8_buffer */
+
+/**
+ * Convert ecma-string's contents to a cesu-8 string and put it to the buffer.
+ * It is the caller's responsibility to make sure that the string fits in the buffer.
+ * Check if the size of the string is equal with the size of the buffer.
+ */
+void __attr_always_inline___
+ecma_string_to_utf8_bytes (const ecma_string_t *string_desc_p, /**< ecma-string descriptor */
+                           lit_utf8_byte_t *buffer_p, /**< destination buffer pointer
+                                                       * (can be NULL if buffer_size == 0) */
+                           lit_utf8_size_t buffer_size) /**< size of buffer */
+{
+  JERRY_ASSERT (ecma_string_get_size (string_desc_p) == buffer_size);
```

Comment:
I think this assert is unnecessary, since the <= is checked by ecma_string_copy_to_utf8_buffer and size == buffer_size checks the equality.